src/Renderer/TwigTemplateRenderer.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * Copyright (c) 2021 Heimrich & Hannot GmbH
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace HeimrichHannot\TwigSupportBundle\Renderer;
+
+use Contao\Config;
+use Contao\CoreBundle\Framework\ContaoFramework;
+use HeimrichHannot\TwigSupportBundle\Exception\TemplateNotFoundException;
+use HeimrichHannot\TwigSupportBundle\Filesystem\TwigTemplateLocator;
+use Twig\Environment;
+use Twig\Error\LoaderError;
+use Twig\Error\RuntimeError;
+use Twig\Error\SyntaxError;
+
+class TwigTemplateRenderer
+{
+    protected $twig;
+    protected $templateLocator;
+    protected $contaoFramework;
+
+    /**
+     * TwigTemplateRenderer constructor.
+     */
+    public function __construct(Environment $twig, TwigTemplateLocator $templateLocator, ContaoFramework $contaoFramework)
+    {
+        $this->twig = $twig;
+        $this->templateLocator = $templateLocator;
+        $this->contaoFramework = $contaoFramework;
+    }
+
+    /**
+     * @throws \HeimrichHannot\TwigSupportBundle\Exception\TemplateNotFoundException
+     * @throws \Twig\Error\LoaderError
+     * @throws \Twig\Error\RuntimeError
+     * @throws \Twig\Error\SyntaxError                                               Configuration
+     */
+    public function render(string $templateName, array $templateData = [], ?TwigTemplateRendererConfiguration $configuration = null): string
+    {
+        if (!$configuration) {
+            $configuration = new TwigTemplateRendererConfiguration();
+        }
+
+        $buffer = null;
+
+        if (null !== $configuration->getTemplatePath()) {
+            try {
+                $templatePath = $this->templateLocator->getTemplatePath($templateName);
+            } catch (TemplateNotFoundException $e) {
+                if ($configuration->getThrowExceptionOnError()) {
+                    throw $e;
+                }
+                $buffer = '';
+            }
+        } else {
+            $templatePath = $configuration->getTemplatePath();
+        }
+
+        if (null === $buffer) {
+            try {
+                $buffer = $this->twig->render($templatePath, $templateData);
+            } catch (LoaderError | RuntimeError | SyntaxError $e) {
```

Comment:
What if Exceptions are thrown not errors?

Reply:
There errors are also exceptions ;) 
real errors could not be cached. Don't know, why they named them erros, I think it's history :D
I don't want to catch other exceptions as they are not normal in this context and so they should be thrown.